test/templates.go

```diff
@@ -10,64 +10,64 @@ import (
 // a new Vault instance and stores it in storage.
 // balance is an argument to the Vault constructor.
 // The Vault must have been deployed already.
-func GenerateCreateTokenScript(fungibleAddr, flowAddr flow.Address) []byte {
+func GenerateCreateTokenScript(fungibleAddr, tokenAddr flow.Address, tokenName, storageName string) []byte {
 	template := `
 	  import FungibleToken from 0x%s 
-	  import FlowToken from 0x%s
+	  import %s from 0x%s
 
       transaction {
 
           prepare(acct: AuthAccount) {
-              let vault <- FlowToken.createEmptyVault()
-              acct.save(<-vault, to: /storage/flowTokenVault)
+              let vault <- %s.createEmptyVault()
+              acct.save(<-vault, to: /storage/%sVault)
 
-              acct.link<&FlowToken.Vault{FungibleToken.Receiver}>(/public/flowTokenReceiver, target: /storage/flowTokenVault)
-              acct.link<&FlowToken.Vault{FungibleToken.Balance}>(/public/flowTokenBalance, target: /storage/flowTokenVault)
+              acct.link<&%s.Vault{FungibleToken.Receiver}>(/public/%sReceiver, target: /storage/%sVault)
+              acct.link<&%s.Vault{FungibleToken.Balance}>(/public/%sBalance, target: /storage/%sVault)
           }
       }
     `
-	return []byte(fmt.Sprintf(template, fungibleAddr, flowAddr))
+	return []byte(fmt.Sprintf(template, fungibleAddr, tokenName, tokenAddr, tokenName, storageName, tokenName, storageName, storageName, tokenName, storageName, storageName))
```

Comment:
You can simplify this a bit with indexes to avoid repeating arguments:

```go
func GenerateCreateTokenScript(fungibleAddr, tokenAddr flow.Address, tokenName, storageName string) []byte {
	template := `
	  import FungibleToken from 0x%[1]s 
	  import %[3]s from 0x%[2]s
      transaction {
          prepare(acct: AuthAccount) {
              let vault <- %[3]s.createEmptyVault()
              acct.save(<-vault, to: /storage/%[4]sVault)
              acct.link<&%[3]s.Vault{FungibleToken.Receiver}>(/public/%[4]sReceiver, target: /storage/%[4]sVault)
              acct.link<&%[3]s.Vault{FungibleToken.Balance}>(/public/%[4]sBalance, target: /storage/%[4]sVault)
          }
      }
    `
	return []byte(fmt.Sprintf(template, fungibleAddr, tokenAddr, tokenName, storageName))
}
```

Reply:
Same for all the templates in this file